tasks/icinga2-bootstrap.yml

```diff
@@ -0,0 +1,125 @@
+---
+# When a --limit is in place only the facts of
+# the selected hosts are retrieved and we need
+# the facts of the other nodes as they are
+# used as defaults of variables used in the
+# templates
+- name: Gather Facts of the other Icinga nodes
```

Comment:
This step will make the entire run fail if not all the hosts in the inventory are reachable.
Could you maybe explain why you added this (perhaps because you had to run using `--limit` for a specific reason)?
Maybe we can find a better solution than adding a `ignore_errors: yes`.

Reply:
We often apply the playbook to a subset of the systems for instance only on the master nodes.

When you do that you don't have the default gather facts actions for the other nodes such as the satellites and as such the configuration generated becomes invalid or can't be generated because you missing infomation. The best way to fix that is to do an setup action on the nodes via delegate_to so all the necessary information is there.

I guess that if some nodes are unreachable the playbook will fail, but that is better than corrupting the configuration :)

Maybe we can restrict this to the configuration master only? But I guess other nodes also need information from neighbouring nodes (satellites in the same zone, information about the nodes in the parent zone) so that won't be easy.

Other option is to make this an configuration flag?

Reply:
Huh… I had sent a reply by email but apparently it never made its way here…

Reply:
This is a tough one, because the user might expect a range of things:

# Some hosts are unreachable (because they have been removed from the
infrastructure) so I also want them remove --> OK, then they should no
longer be in the inventory

# Some hosts are unreachable (but it's a temporary thing) --> In this
case, you'd probably want their config to remain untouched. You'd set
`i2_remove_unmanaged_objects` to `false`.

# You run your playbook on a subset of hosts on purpose (because you
don't want ansible to check again whether icinga2 is installed, etc…)
--> In this case, the cases above could actually still apply. Some of
the hosts from which you'd like to gather facts could be unreachable.


What I've been doing mostly is use the tags to make runs as quick as
possible. E.g. when I use a `--tags "objects"` only, then this comes
pretty close to a `--limit` run on just the configuration master. Tasks
tagged "always" are still run on all reachable hosts, but then
configuration is only generated on the configuration master.